pyproject.toml

```
@@ -0,0 +1,132 @@
+[project]
+name = "paperless-ngx"
+version = "0.1.0"
+description = "Add your description here"
+readme = "README.md"
+requires-python = ">=3.10"
+classifiers = [
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+  "bleach~=6.2.0",
+  "celery[redis]~=5.4.0",
+  "channels~=4.2",
+  "channels-redis~=4.2",
+  "concurrent-log-handler~=0.9.25",
+  "dateparser~=1.2",
+  # WARNING: django does not use semver.
+  #          Only patch versions are guaranteed to not introduce breaking changes.
+  "django~=5.1.6",
+  "django-allauth[socialaccount,mfa]~=65.4.0",
+  "django-auditlog~=3.0.0",
+  "django-celery-results~=2.5.1",
+  "django-compression-middleware~=0.5.0",
+  "django-cors-headers~=4.7.0",
+  "django-extensions~=3.2.3",
+  "django-filter~=25.1",
+  "django-guardian~=2.4.0",
+  "django-multiselectfield~=0.1.13",
+  "django-soft-delete~=1.0.18",
+  "djangorestframework~=3.15",
+  "djangorestframework-guardian~=0.3.0",
+  "drf-spectacular~=0.28",
+  "drf-spectacular-sidecar~=2025.2.1",
+  "drf-writable-nested~=0.7.1",
+  "filelock~=3.17.0",
+  "flower~=2.0.1",
+  "gotenberg-client~=0.9.0",
+  "httpx-oauth~=0.16",
+  "imap-tools~=1.10.0",
+  "inotifyrecursive~=0.3",
+  "jinja2~=3.1.5",
+  "langdetect~=1.0.9",
+  "mysqlclient~=2.2.7",
+  "nltk~=3.9.1",
+  "ocrmypdf~=16.9.0",
+  "pathvalidate~=3.2.3",
+  "pdf2image~=1.17.0",
+  "psycopg[c]==3.2.4",
+  "python-dateutil~=2.9.0",
+  "python-dotenv~=1.0.1",
+  "python-gnupg~=0.5.4",
+  "python-ipware~=3.0.0",
+  "python-magic~=0.4.27",
+  "pyzbar~=0.1.9",
+  "rapidfuzz~=3.12.1",
+  "redis[hiredis]~=5.2.1",
+  "scikit-learn~=1.6.1",
+  "setproctitle~=1.3.4",
+  "tika-client~=0.9.0",
+  "tqdm~=4.67.1",
+  "watchdog~=6.0",
+  "whitenoise~=6.9",
+  "whoosh~=2.7",
+  "zxing-cpp~=2.3.0",
+]
+
+optional-dependencies.webserver = [
+  "granian~=1.7.6",
+]
```

This would also be a great opportunity to make mysql and postgres optional dependencies that can be installed. Both packages require being built from source and require mysql and postgres being installed on the local system. You could then still get it with the extra.

```suggestion
[project.optional-dependencies]
webserver = ["granian~=1.7.6",]
postgres = ["psycopg[c]==3.2.4",]
mysql = ["mysqlclient~=2.2.7",]
```

---

I plan to, but it might be a second pr. This is already getting large